t/14-grutasks-git.t

```diff
@@ -448,14 +452,102 @@ subtest 'delete_needles' => sub {
         $res = run_gru_job(@gru_args);
         is $res->{state}, 'finished', 'job finished';
         $error = $res->{result}->{errors}->[0];
         like $error->{message}, qr{Another git task for.*fedora.*is ongoing}, 'expected error message';
     };
 };
 
+subtest ServerAvailability => sub {
+    my $tmpdir = tempdir();
+    local $ENV{OPENQA_GIT_SERVER_OUTAGE_FILE} = "$tmpdir/mock";
+
+    subtest 'no flag file => skip' => sub {
+        my $outcome = report_server_unavailable($t->app, 'gitlab');
+        is $outcome, 'SKIP', 'Skips when flag file for gitlab is absent';
+    };
+
+    subtest 'file present but younger than 1800 => skip' => sub {
+        my $mock_file = $tmpdir->child('mock.gitlab.flag')->touch;
+        my $outcome = report_server_unavailable($t->app, 'gitlab');
+        is $outcome, 'SKIP', 'Skips the job for an mtime < 1800';
+    };
+
+    subtest 'file present and older => fail' => sub {
+        my $mock_file = $tmpdir->child('mock.gitlab.flag')->touch;
+        my $old_time = time - 2000;
+        utime($old_time, $old_time, $mock_file)
+          or die "Couldn't change mtime on file: $mock_file - $!";
+        my $outcome = report_server_unavailable($t->app, 'gitlab');
+        is $outcome, 'FAIL', 'Fails the job when mtime >= 1800';
+    };
+
+    subtest 'report_server_available removes existing flag file' => sub {
+        my $mock_file = $tmpdir->child('mock.gitlab.flag')->touch;
+        ok -f $mock_file, 'mock.gitlab.flag exists initially';
+        report_server_available($t->app, 'gitlab');
+        ok !-f $mock_file, 'gitlab flag file was removed';
+    };
+
+    subtest 'multiple servers do not conflict' => sub {
+        my $outcome_gitlab = report_server_unavailable($t->app, 'gitlab');
+        is $outcome_gitlab, 'SKIP', 'No gitlab flag => skip';
+        my $gitlab_file = "$tmpdir/mock.gitlab.flag";
+        ok -f $gitlab_file, 'Created mock.gitlab.flag';
+        my $outcome_github = report_server_unavailable($t->app, 'github');
+        is $outcome_github, 'SKIP', 'No github flag => skip';
+        my $github_file = "$tmpdir/mock.github.flag";
+        ok -f $github_file, 'Created mock.github.flag';
+
+        my $old_time = time - 2000;
+        utime($old_time, $old_time, $github_file);
+        # Re-check
+        $outcome_gitlab = report_server_unavailable($t->app, 'gitlab');
+        $outcome_github = report_server_unavailable($t->app, 'github');
+        is $outcome_gitlab, 'SKIP', 'GitLab file still fresh => skip';
+        is $outcome_github, 'FAIL', 'GitHub file older => fail';
+    };
+
+    subtest Clone => sub {
+        my $clone_mock = Test::MockModule->new('OpenQA::Task::Git::Clone');
+        $clone_mock->redefine(_git_clone => sub { die "Internal API unreachable\n" });
+        subtest 'Internal API unreachable => skip' => sub {
+            my @gru_args = (
+                $t->app,
+                'git_clone',
+                {
+                    priority => 10,
```

Comment:
What does the priority have to do here?
This should be a hash with directory names as keys and urls (or undef) as values.

Reply:
sorry i assumed this was needed as all other gru_args defintions within the file had this set as well.